test_scripts/API/VehicleData/WindowStatus/common.lua

```diff
@@ -0,0 +1,454 @@
+  :Times(m.getAppsCount())
+  local isSDLShutDownSuccessfully = false
+  m.getHMIConnection():ExpectNotification("BasicCommunication.OnSDLClose")
+  :Do(function()
+      utils.cprint(35, "SDL was shutdown successfully")
+      isSDLShutDownSuccessfully = true
+      RAISE_EVENT(event, event)
+    end)
+  :Timeout(timeout)
+  local function forceStopSDL()
+    if isSDLShutDownSuccessfully == false then
+      utils.cprint(35, "SDL was shutdown forcibly")
+      RAISE_EVENT(event, event)
+    end
+  end
+  RUN_AFTER(forceStopSDL, timeout + 500)
+end
+
+-- [[ @unexpectedDisconnect: closing connection
+-- ! @parameters: none
+-- ! @return: none
+-- ]]
+function m.unexpectedDisconnect()
+  m.getHMIConnection():ExpectNotification("BasicCommunication.OnAppUnregistered", { unexpectedDisconnect = true })
+  :Times(actions.mobile.getAppsCount())
+  actions.mobile.disconnect()
+  utils.wait(1000)
+end
+
+--[[ @checkResumption_FULL: function that checks HMIlevel to FULL recovery after resumption
+--! @parameters: none
+--! @return: none
+--]]
+function m.checkResumption_FULL()
+  m.getHMIConnection():ExpectRequest("BasicCommunication.ActivateApp", {})
+  :Do(function(_, data)
+    m.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+  end)
+  m.getMobileSession():ExpectNotification("OnHMIStatus",
+    { hmiLevel = "NONE" },
+    { hmiLevel = "FULL" })
+  :Times(2)
+end
+
+--[[ @checkResumption_NONE: function that checks HMIlevel to NONE recovery after resumption
+--! @parameters: none
+--! @return: none
+--]]
+function m.checkResumption_NONE()
+  m.getMobileSession(1):ExpectNotification("OnHMIStatus",{ hmiLevel = "NONE" })
+end
+
+
+--[[ @registerWithResumption: re-register application with SUCCESS resultCode
+--! @parameters:
+--! pAppId: application number (1, 2, etc.)
+--! pLevelCheckFunc: check function
+--! isHMIsubscription: if false SDL resumes SubscribeVehicleData for app2 and does not send VehicleInfo.SubscribeVehicleData request to HMI
+--! @return: none
+--]]
+function m.registerWithResumption(pAppId, pLevelCheckFunc, isHMIsubscription)
+  if not pAppId then pAppId = 1 end
+  m.getMobileSession(pAppId):StartService(7)
+  :Do(function()
+    local params = m.cloneTable(m.getConfigAppParams(pAppId))
+    params.hashID = m.getHashId(pAppId)
+    local corId = m.getMobileSession(pAppId):SendRPC("RegisterAppInterface", params)
+    m.getHMIConnection():ExpectNotification("BasicCommunication.OnAppRegistered", {
+      application = { appName = m.getConfigAppParams(pAppId).appName }
+    })
+    :Do(function(_, data)
+      if true == isHMIsubscription then
+        m.getHMIConnection():ExpectRequest( "VehicleInfo.SubscribeVehicleData", { windowStatus = true })
+        m.getHMIConnection():SendResponse( data.id, data.method, "SUCCESS", { windowStatus = m.subUnsubParams })
+      else
+        m.getHMIConnection():ExpectRequest( "VehicleInfo.SubscribeVehicleData", { windowStatus = true })
+        :Times(0)
```

Comment:
For `0` expectations we don't need to define expected result (`{ windowStatus = true }`)